mPower2/mPower2/View Controllers/TaskBrowserViewController.swift

```diff
@@ -0,0 +1,366 @@
+        collectionView.reloadData()
+        
+        // Now update the isSelected value of all the tabs
+        tabButtonStackView.arrangedSubviews.forEach({
+            if let tabView = $0 as? TaskBrowserTabView {
+                tabView.isSelected = tabView.taskGroupIdentifier == identifier
+            }
+        })
+    }
+}
+
+extension TaskBrowserViewController: UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
+    
+    // MARK: UICollectionViewDataSource
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        guard let selectedTaskGroup = selectedTaskGroup else {
+            return 0
+        }
+        return selectedTaskGroup.tasks.count
+    }
+    
+    // MARK: UICollectionViewDelegate
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: kCollectionCellIdentifier, for: indexPath) as? TaskCollectionViewCell
+        if let selectedTaskGroup = selectedTaskGroup {
+            
+            let task = selectedTaskGroup.tasks[indexPath.row]
+            cell?.image = nil
+            task.imageVendor?.fetchImage(for: CGSize(width: 0.0, height: 0.0)) { (_, img) in
+                cell?.image = img
+            }
+            cell?.title = task.title?.uppercased()
+        }
+        return cell ?? UICollectionViewCell()
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return cellSize
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        
+        guard let selectedTaskGroup = selectedTaskGroup else {
+            return
+        }
+
+        // Get our task and present it
+        let taskInfo = selectedTaskGroup.tasks[indexPath.row]
+        guard let taskPath = selectedTaskGroup.instantiateTaskPath(for: taskInfo) else { return }
+        let vc = RSDTaskViewController(taskPath: taskPath)
+        vc.delegate = self
+        self.present(vc, animated: true, completion: nil)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, insetForSectionAt section: Int) -> UIEdgeInsets {
+        
+        guard let selectedTaskGroup = selectedTaskGroup else {
+            return UIEdgeInsetsMake(0, 0, 0, 0)
+        }
+        
+        // We do this so the cells are centered horizontally
+
+        let totalCellWidth = cellSize.width * CGFloat(selectedTaskGroup.tasks.count)
+        let totalSpacingWidth = minCellSpacing * (CGFloat(selectedTaskGroup.tasks.count) - 1)
+        let inset = (collectionView.frame.size.width - CGFloat(totalCellWidth + totalSpacingWidth)) / 2
```

Comment:
I would prefer to change the spacing rather than the insets and also do the math for a case where there are 4 or 5 cells on an iPhone SE.

Reply:
I added a todo for this